### PR TITLE
Added sigint_timeout to drivers that call kill_process

### PR DESCRIPTION
**Pull request by nakjemmyMS**
## Bug / Requirement Description
Clearly and concisely describe the problem.

## Solution description
Describe your code changes in detail for reviewers.

## Checklist:
- [ ] Test
- [ ] Example (both test_plan.py and .rst)
- [ ] Documentation (API)
- [ ] News fragment present for release notes
- [ ] MS info leakage check
- [ ] For new driver: driver index page
- [ ] For new assertion: ui/pdf/std renderers, documentation
- [ ] For new cmdline arg: documentation

### CODE REVIEW

File: testplan/common/remote/remote_service.py [hunk 1]
```diff
@@ -60,6 +61,8 @@ class RemoteService(Resource, RemoteResource):
     :param rpyc_port: Specific port for rpyc connection on the remote host. Defaults to 0
         which start the rpyc server on a random port.
     :type rpyc_port: ``int``
+    :param sigint_timeout: number of seconds to wait between ``SIGINT`` and ``SIGKILL``
+    :type sigint_timeout: ``int``
```
Comment (M6AI) [inline testplan/common/remote/remote_service.py:65]: No need to add `:type ...:` directive when the type annotation is in place in the function signature. Can you please remove all of the `:type`-stuffy in the docstring here? I see all the parameters annotated.

File: testplan/runners/pools/process.py [hunk 2]
```diff
@@ -43,12 +46,22 @@ class ProcessWorker(Worker):
 
     :param transport: Transport class for pool/worker communication.
     :type transport: :py:class:`~testplan.runners.pools.connection.Client`
+    :param sigint_timeout: number of seconds to wait between ``SIGINT`` and ``SIGKILL``
+    :type sigint_timeout: ``int``
```
Comment (M6AI) [inline testplan/runners/pools/process.py:50]: Same here.

File: testplan/testing/multitest/driver/app.py [hunk 3]
```diff
@@ -96,6 +97,8 @@ class App(Driver):
     :param expected_retcode: the expected return code of the subprocess.
         Default value is None meaning it won't be checked. Set it to 0 to
         ennsure the driver is always gracefully shut down.
+    :param sigint_timeout: number of seconds to wait between ``SIGINT`` and ``SIGKILL``
+    :type sigint_timeout: ``int``
```
Comment (M6AI) [inline testplan/testing/multitest/driver/app.py:101]: Same here.

**Review by M6AI — CHANGES_REQUESTED** [review]

**Comment by M6AI** [comment]
In addition, please provide a newsfragment.

File: testplan/common/remote/remote_service.py (hunk 1 above)
Comment (M6AI) [reply testplan/common/remote/remote_service.py:65]: It is, but it is our old school doc approach. :type: duplicates content on API docs (that is already not the nicest thing to read) and does not support intellisense as seamless as type annotations.

File: doc/newsfragments/2412_new.customizable_sigint_timeout_for_drivers.rst [hunk 4]
```diff
@@ -0,0 +1 @@
+Added sigint_timeout to drivers that call kill_process
```
Comment (M6AI) [inline doc/newsfragments/2412_new.customizable_sigint_timeout_for_drivers.rst:1]: Some reformatting is due, please enclose sigint_timeout and kill process in ````

Comment (M6AI) [inline doc/newsfragments/2412_new.customizable_sigint_timeout_for_drivers.rst:1]:
```suggestion
Added sigint_timeout to `drivers` that call `kill_process`
```

Comment (nakjemmyMS) [reply doc/newsfragments/2412_new.customizable_sigint_timeout_for_drivers.rst:1]: Okay sure, well noted

**Review by M6AI — DISMISSED** [review]

**Review by M6AI — APPROVED** [review]

**Pull request merged**